### ISSUE TITLE
Failed upgrade due to filter box migration

### ISSUE THREAD

**Issue by sadpandajoe** [apache/superset#27885]
### Bug description

Failed upgrade while trying to remove filter box.

### How to reproduce the bug

We had an older account that we tried to upgrade to latest version of 4.0 and ran into this error:

> update or delete on table "slices" violates foreign key constraint "report_schedule_chart_id_fkey" on table "report_schedule" DETAIL: Key (id)=(155) is still referenced from table "report_schedule". [SQL: DELETE FROM slices WHERE slices.viz_type = %(viz_type_1)s] [parameters: {'viz_type_1': 'filter_box'}

### Superset version

master / latest-dev

### Python version

3.10

### Node version

16

### Browser

Chrome

### Checklist

- [X] I have searched Superset docs and Slack and didn't find a solution to my problem.
- [X] I have searched the GitHub issue tracker and didn't find a similar bug report.
- [ ] I have checked Superset's logs for errors and if I found a relevant Python stacktrace, I included it here as text in the "additional context" section.

**Comment by michael-s-molina** [comment]
@sadpandajoe @john-bodley I'm not sure if we should consider this as a bug. This error will only happen if you have a filter box as the selected chart for an alert/report which does not make any sense. I believe this seems unlikely and should be treated as corrupted data and cleaned manually. Do you agree?

**Comment by john-bodley** [comment]
@michael-s-molina I agree. It seems somewhat atypical to have a filter box as an alert or report. For such cases it’s likely best that the exception is thrown, i.e., reaching out to the user to understand what they were trying to achieve and having them reformulate the alert or report using an actual chart.

**Comment by sadpandajoe** [comment]
@michael-s-molina @john-bodley agreed, it was just something we have run into while doing a 4.0 upgrade and easily able to handle by removing that alert. Just posting it for an audit trail in case anyone runs into this problem during their upgrade.

**Comment by michael-s-molina** [comment]
@sadpandajoe Thanks for testing the release and opening the issue as it might help others if they find the same problem.
